pkg/ui/workspaces/cluster-ui/src/store/jobDetails/job.selectors.ts

```diff
@@ -9,9 +9,17 @@
 // licenses/APL.txt.
 
 import { createSelector } from "reselect";
-import { adminUISelector } from "../utils/selectors";
+import { adminUISelector } from "src/store/utils/selectors";
+import { selectID } from "src/selectors";
 
 export const selectJobState = createSelector(
   adminUISelector,
-  adminUiState => adminUiState.job,
+  selectID,
+  (state, jobID) => {
+    const jobCache = state?.job?.cachedData;
```

Comment:
Is the job data type being cached here the same as what's being stored in the `jobs` list? If so, maybe we can avoid an extra request if we already have the job we're trying to view in the `jobs` list. Maybe we can look in that list first and put that in the cache if there's a result.